flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontend.java

```diff
@@ -735,6 +829,116 @@ private void disposeSavepoint(ClusterClient<?> clusterClient, String savepointPa
 		logAndSysout("Savepoint '" + savepointPath + "' disposed.");
 	}
 
+	/**
+	 * The '-h' or '--help' action.
+	 */
+	protected class ActionHelp implements CommandAction<CommandLineOptions> {
+
+		@Override
+		public CommandLineOptions parse(String[] args) throws CliArgsException {
```

Comment:
@V1ncentzzZ please check this out to help you understand why I removed `abstract` of `CommandLineOptions`.
This implementation needs to return an instance of `CommandLineOptions`, but we do not have a proper type of `CommandLineOptions` to create.